frontend/src/features/frame/components/topbar/AccountInfo.tsx

```diff
@@ -30,14 +34,28 @@ const useStyles = makeStyles({
 });
 
 const LoggedInActions = () => {
+  const { enqueueSnackbar } = useSnackbar();
+
   const classes = useStyles();
   const { logout, loginState } = useLoginState();
 
+  const logoutProcess = async () => {
+    if (loginState.access_token) {
+      await revokeAccessToken(loginState.access_token).catch(() => {
```

Comment:
As far as I can see, the refresh_token remains valid after the access_token has been revoked. I think we would expect both to be revoked, to avoid any credentials from a previous session to be reused.

Calling `/o/revoke_token` with the refresh_token would actually be enough, as the API will take care of deleting the associated access_token.

Reply:
> As far as I can see, the refresh_token remains valid after the access_token has been revoked.

That's true.

> I think we would expect both to be revoked, to avoid any credentials from a previous session to be reused.

You're right, for a reason I thought the `/o/revoke_token/` was only for access tokens.

Reply:
According to the [RFC 7009 section 2.1](https://datatracker.ietf.org/doc/html/rfc7009#section-2.1), revoking a refresh token " should " also revoke the access token.

>  If the particular token is a refresh token and the authorization server supports the revocation of access tokens, then the authorization server SHOULD also invalidate all access tokens based on the same authorization grant (see Implementation Note).  If the token passed to the request is an access token, the server MAY revoke the respective refresh token as well.

But, Django OAuth Toolkit seems to only revoke the access token, even if I use the refresh token in the request.

Example, I expected the following request to delete the refresh token `75U5OZPTCeXC2jnfpkj2qHJz6mNg4y`, but it deletes the related access token instead.

```http
POST /o/revoke_token/ HTTP/1.1
Content-Type: application/x-www-form-urlencoded
token=75U5OZPTCeXC2jnfpkj2qHJz6mNg4y&client_id=[...]&client_secret=[...]&token_type_hint=refresh_token
```

I'm gonna do more tests.

Reply:
> Example, I expected the following request to delete the refresh token 75U5OZPTCeXC2jnfpkj2qHJz6mNg4y, but it deletes the related access token instead.

 The refresh_token is also revoked, but contrary to the access_token it's not deleted from the database immediately. It's only marked as revoked, via the field `revoked`.

Reply:
Thanks!

Reply:
Fixed by d63b49c10f9bb4ef67522571e7ed04d231069a66.